src/data/player.py

```diff
@@ -282,7 +282,13 @@ def year_of_birth(self) -> int:
 
     @cached_property
     def category(self) -> PlayerCategory:
-        return PlayerCategory.from_year_of_birth(self.year_of_birth)
+        assert self.tournament is not None
+        event = self.tournament.event
+        return PlayerCategory.from_year_of_birth(
+            self.year_of_birth,
+            event.start_datetime,
+            event.stop_datetime,
+        )
```

Comment:
Most of the time all the tournaments of an event have the sames start/end dates.
Anyway, there may be cases where players have different catégories in different tournaments, e.g. for special 'Grand Prix' events.
```suggestion
        assert self.tournament is not None
        return PlayerCategory.from_year_of_birth(
            self.year_of_birth,
            tournament.start_datetime,
            tournament.stop_datetime,
        )
```
(add ``Tournament.xxx_datetime()`` methods)